cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -13,29 +13,93 @@
 # This module contains all the classes representing the Schema of the configuration file.
 # These classes are created by following marshmallow syntax.
 #
-
-from marshmallow import Schema, ValidationError, fields, post_dump, post_load, pre_load
+import re
+
+from marshmallow import (
+    Schema,
+    ValidationError,
+    fields,
+    post_dump,
+    post_load,
+    pre_dump,
+    pre_load,
+    validate,
+    validates,
+    validates_schema,
+)
 
 from pcluster.config.cluster_config import (
+    AdditionalIamPolicyConfig,
+    BaseEbsConfig,
+    CloudWatchDashboardsConfig,
+    CloudWatchLogsConfig,
     ClusterConfig,
     ComputeResourceConfig,
+    CustomActionConfig,
+    DashboardsConfig,
+    DcvConfig,
     EbsConfig,
     EfsConfig,
+    EphemeralVolumeConfig,
     FsxConfig,
     HeadNodeConfig,
     HeadNodeNetworkingConfig,
+    IamConfig,
     ImageConfig,
+    LogsConfig,
+    MonitoringConfig,
+    PlacementGroupConfig,
+    ProxyConfig,
     QueueConfig,
     QueueNetworkingConfig,
+    RaidConfig,
+    RolesConfig,
+    S3AccessConfig,
     SchedulingConfig,
-    SharedStorageType,
+    SchedulingSettingsConfig,
+    SharedStorageConfig,
     SshConfig,
+    StorageConfig,
+    TagConfig,
 )
+from pcluster.constants import FSX_HDD_THROUGHPUT, FSX_SSD_THROUGHPUT, SUPPORTED_ARCHITECTURES
+
+ALLOWED_VALUES = {
+    "cidr": r"^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}"
+    r"([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])"
+    r"(\/([0-9]|[1-2][0-9]|3[0-2]))$",
+    "file_path": r"^\/?[^\/.\\][^\/\\]*(\/[^\/.\\][^\/]*)*$",
+    "security_group_id": r"^sg-[0-9a-z]{8}$|^sg-[0-9a-z]{17}$",
+    "subnet_id": r"^subnet-[0-9a-z]{8}$|^subnet-[0-9a-z]{17}$",
+    "architectures": SUPPORTED_ARCHITECTURES,
+}
+
+
+def _get_field_validator(field_name):
+    allowed_values = ALLOWED_VALUES[field_name]
+    return validate.OneOf(allowed_values) if isinstance(allowed_values, list) else validate.Regexp(allowed_values)
+
+
+def _camelcase(s):
+
+    parts = iter(s.split("_"))
+    return "".join(i.title() for i in parts)
 
 
 class BaseSchema(Schema):
     """Represent a base schema, containing all the features required by all the Schema classes."""
 
+    def on_bind_field(self, field_name, field_obj):
```

Comment:
This is a great improvement.

Reply:
Thank you. I was desperate before having this method. Haha